VideoLocker/src/main/java/org/edx/mobile/view/CourseOutlineActivity.java

```diff
@@ -38,67 +38,63 @@ protected void onCreate(Bundle savedInstanceState) {
-            CourseComponent courseComponent = courseManager.getComponentById(courseData.getCourse().getId(), courseComponentId);
-            if ( courseComponent == null)
-                setTitle( courseData.getCourse().getName() );
-            else
-                setTitle( courseComponent.getName() );
-
-            if (isOnCourseOutline())
-                LastAccessManager.getSharedInstance().fetchLastAccessed(this, courseData.getCourse().getId());
+        setTitle(courseData.getCourse().getName());
+
+        if (isOnCourseOutline())
+            LastAccessManager.getSharedInstance().fetchLastAccessed(this, courseData.getCourse().getId());
+    }
+
+    @Override
+    protected void onLoadData() {
+        CourseComponent courseComponent = courseManager.getComponentById(
+                courseData.getCourse().getId(), courseComponentId);
+        setTitle(courseComponent.getName());
+
+        if (fragment == null) {
+            fragment = new CourseOutlineFragment();
+            fragment.setTaskProcessCallback(this);
+
+            Bundle bundle = new Bundle();
+            bundle.putSerializable(Router.EXTRA_ENROLLMENT, courseData);
+            bundle.putString(Router.EXTRA_COURSE_COMPONENT_ID, courseComponentId);
+            bundle.putString(Router.EXTRA_LAST_ACCESSED_ID
+                    , getIntent().getStringExtra(Router.EXTRA_LAST_ACCESSED_ID));
+            fragment.setArguments(bundle);
+            //this activity will only ever hold this lone fragment, so we
+            // can afford to retain the instance during activity recreation
+            fragment.setRetainInstance(true);
+
+            FragmentTransaction fragmentTransaction = getSupportFragmentManager().beginTransaction();
+            fragmentTransaction.add(R.id.fragment_container, fragment, CourseOutlineFragment.TAG);
+            fragmentTransaction.disallowAddToBackStack();
+            fragmentTransaction.commit();
+        }
+
+        if (isOnCourseOutline()) {
+            LastAccessManager.getSharedInstance().fetchLastAccessed(this, courseData.getCourse().getId());
+        } else {
+            // Update the last accessed item reference if we are in the course subsection view
+            String prefName = PrefManager.getPrefNameForLastAccessedBy(getProfile()
+                    .username, courseComponent.getCourseId());
+            final PrefManager prefManager = new PrefManager(MainApplication.instance(), prefName);
+            prefManager.putLastAccessedSubsection(courseComponent.getId(), false);
         }
     }
 
     @Override
     protected String getUrlForWebView() {
-        if ( courseData == null )
-            return "";
+        if ( courseComponentId == null ) return "";
         CourseComponent courseComponent = courseManager.getComponentById(courseData.getCourse().getId(), courseComponentId);
-        return courseComponent == null ? "" : courseComponent.getWebUrl();
```

Comment:
This should probably be returning the webURL for the current block not the course?